src/main/java/com/griddynamics/service/ProductService.java

```diff
@@ -45,12 +50,16 @@ public ProductDTO getById(Integer id) throws ServiceException {
 
         Optional<Product> optProduct = productRepository.findById(id);
 
-        ProductDTO productDTO = null;
+        ProductDTO productDTO;
 
         if (optProduct.isPresent()) {
             productDTO = productMapper.mapDTO(optProduct.get());
```

Comment:
productDTO = productMapper.mapDTO(optProduct.orElseThrow(new ProductNotFoundException("Product is not exist.");));